### PR TITLE
update docstrings

### PR DESCRIPTION
**Pull request by lucaferranti**
- [x] utils.jl
- [x] rref.jl
- [x] classify.jl

### CODE REVIEW

File: src/classify.jl [hunk 1]
```diff
@@ -1,11 +1,29 @@
 # Routines to classify interval matrices
 
 """
-    is_strongly_regular(A)
+    is_strongly_regular(A::AbstractMatrix{T}) where {T<:Interval}
```
Comment (mforets) [inline src/classify.jl:4]: the docstring doesn't match the functions' signature

same below

File: src/classify.jl [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
 
-Tests whether an interval matrix A is strongly regular.
-An interval matrix `A` is strongly regular if Ac⁻¹A is regular.
-For more details see section 4.6 of [[HOR19]](@ref)
+Tests whether an interval matrix ``A`` is strongly regular, i.e. if if ``A_c^{-1}A`` is
+an H-matrix. For more details see section 4.6 of [[HOR19]](@ref).
+
+### Examples
```
Comment (mforets) [inline src/classify.jl:9]:
```suggestion
### Examples

```
an empty line here, after the title, improves the readability

File: src/classify.jl [hunk 3]
```diff
@@ -1,11 +1,29 @@
 # Routines to classify interval matrices
 
 """
-    is_strongly_regular(A)
+    is_strongly_regular(A::AbstractMatrix{T}) where {T<:Interval}
 
-Tests whether an interval matrix A is strongly regular.
-An interval matrix `A` is strongly regular if Ac⁻¹A is regular.
-For more details see section 4.6 of [[HOR19]](@ref)
+Tests whether an interval matrix ``A`` is strongly regular, i.e. if if ``A_c^{-1}A`` is
```
Comment (mforets) [inline src/classify.jl:6]:
```suggestion
Tests whether an interval matrix ``A`` is strongly regular, i.e. if ``A_c^{-1}A`` is
```

File: src/classify.jl [hunk 4]
```diff
@@ -35,11 +72,30 @@ end
...
 
 """
-    is_strictly_diagonally_dominant(A)
-
-Checks whether an interval matrix A is stictly diagonally dominant, that is
-if mig(Aᵢᵢ) > ∑_(k ≠ i) mag(Aᵢₖ) for i=1,…,n.
-For more details see section 4.5 of [[HOR19]](@ref)
+    is_strictly_diagonally_dominant(A::AbstractMatrix{T}) where {T<:Interval}
+
+Checks whether an interval matrix ``A`` is stictly diagonally dominant, that is
```
Comment (mforets) [inline src/classify.jl:77]: ... a square matrix ``A`` of order ``n`` ...

File: src/classify.jl (hunk 1 above)
Comment (lucaferranti) [reply src/classify.jl:4]: updated!

File: src/classify.jl (hunk 2 above)
Comment (lucaferranti) [reply src/classify.jl:10]: updated!

File: src/classify.jl (hunk 4 above)
Comment (lucaferranti) [reply src/classify.jl]: updated here and in the other functions, now they all read "square interval matrix"

**Pull request merged**